### ISSUE TITLE
Hint mode and scroll mode key sequence triggers should work for exiting and switching between modes

### ISSUE THREAD

**Issue by blackketter** [dexterleng/vimac#385]
I use the key sequence triggers to activate Vimac.  

It would be great if the configured key sequences worked while Vimac is active to switch between modes and deactivate modes, rather than be assigned to hints.

So, using the defaults the defaults, if I type `f` `d` to trigger hint mode, hitting `f` `d` again would exit hint mode, and `j` `k` would switch to scroll mode.

**Comment by al3xandru** [comment]
Came here to kindly request the same behavior. Moving to the opposite side of the keyboard to press escape is not ideal.

**Comment by dexterleng** [comment]
I decided against doing this because `f` is part of the default hint keys

**Comment by dexterleng** [comment]
@blackketter @al3xandru 

Could you share some insight into why you prefer to use key sequence over shortcut activation?

**Comment by dexterleng** [comment]
I've thought about having a hold key to activate Vimac. e.g. hold `f` to activate hint-mode. This solution would have a lower change of false-positives and can be used to toggle on/off modes without hint keys conflict issue.

**Comment by blackketter** [comment]
> Could you share some insight into why you prefer to use key sequence over shortcut activation? This would make it easier for me to think of a solution with minimal drawbacks/weird edge cases
For me, it's all about speed.  I use FJ for hint mode, which on a US keyboard are the two index fingers.  They are already on the home row and in alternate hands so I can type the sequence almost as fast as a single key, without moving the fingers. 

(I use JF for scroll mode, so I can choose which mode by pressing the keys in the other order.) 

As a point of interest, I had tried using BetterTouchTool to map a double tap of the left shift key to trigger Vimac, but found the FJ sequence faster.

As far as conflicting with default hint keys, as long as you don't generate hints that are the same as the trigger sequence, it should "just work", right?

**Comment by dexterleng** [comment]
> As far as conflicting with default hint keys, as long as you don't generate hints that are the same as the trigger sequence, it should "just work", right?

Additionally, the hint-text must not end with the first character in the key sequence otherwise there will be a delay

**Comment by colossatr0n** [comment]
@blackketter There's also the option to use CTRL + [ to exit scroll mode instead of ESC. Little less of a stretch and it's a default way to exit mode in vim.

With hint mode, anything that you type that isn't a hint will exit hint mode, so sometimes I'll use a non-hint character to exit such as semicolon.

But I agree, it would be nice to disable either mode by retyping the key sequence.

**Comment by blackketter** [comment]
Right.  What I'm hoping for is the ability to use the same key sequence that triggers hint mode (or scroll mode) to switch to that mode directly.
